{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -7,159 +7,63 @@
-    #    bit of a pain to do now.
-    for location in locations:
-        proper = location.strip()
-        sanitized = re.sub("[- ]", '_', proper).lower()
-        sanitized = sanitized.replace("'", '-')
-        yield Location(proper, sanitized)
-
-
-def parse_locations(locations_file: Optional[str], single_location: Optional[str]) -> List[str]:
-    """Parses location inputs into a list of location strings.
+from .app_logging import configure_logging_to_terminal
+from .make_specs import build_model_specifications
 
-    If no arguments are provided, this will default to the ``locations.txt``
-    file located in the repository model_specifications directory.
 
-    Parameters
-    ----------
-    locations_file
-        Path to a file containing a list of locations to generate model
-        specifications for.
-    single_location
-        Optional single location to generate a model specification for.
-
-    Returns
-    -------
-        A list of location strings.
-
-    Raises
-    ------
-    ValueError
-        If both ``locations_file`` and ``single_loc`` are provided or if
-        a ``locations_file`` with no locations is provided.
-
-    """
-    if locations_file and single_location:
-        raise ValueError('You provided both a locations file and a single location to make_specs.')
-
-    if single_location:
-        return [single_location]
-
-    locations_file = Path(locations_file) if locations_file else MODEL_SPEC_DIR / 'locations.txt'
-    with locations_file.open() as f:
-        # Interpret each line that doesn't start with a '#' as a single location.
-        locations = [l for l in f.readlines() if not l.strip().startswith('#') and l.strip()]
-    if not locations:
-        raise ValueError(f'No locations provided in location file {str(locations_file)}.')
-
-    return locations
-
-
-MODEL_SPEC_DIR = (Path(__file__).parent.parent / 'model_specifications').resolve()
 @click.command()
-@click.option('-l', '--locations-file',
-              type=click.Path(dir_okay=False),
-              help=('The file with the location parameters for the template. If no locations file is provided '
-                    'and no single location is provided with the ``-s`` flag, this will default to the locations '
-                    f'file located at {str(MODEL_SPEC_DIR / "locations.txt")}.'))
 @click.option('-t', '--template',
-              default=str(MODEL_SPEC_DIR / 'model_spec.in'),
+              default=str(paths.MODEL_SPEC_DIR / 'model_spec.in'),
               show_default=True,
               type=click.Path(exists=True, dir_okay=False),
               help='The model specification template file.')
-@click.option('-s', '--single-location',
-              default='',
-              help='Specify a single location name.')
+@click.option('-l', '--location',
+              default='all',
+              show_default=True,
+              type=click.Choice(project_globals.LOCATIONS + ['all']),
```

Comment:
Is `project_globals.LOCATION + ['all']` used anywhere else in the template? If so, should that be stored explicitly in `project_globals` as well? 

Also, should we have `'all'` stored as a constant?

Reply:
'all' isn't really global, it's a marker that only this function recognizes. We have LOCATION as to-be-done by user, but I think it would be kind of weird to expect them to include all. I think 'all' makes sense here